### PR TITLE
Implement get music

### PR DESCRIPTION
**Pull request by retrouser955**
Implement the scraping of music data from the 'youtube.com/watch' domain

### CODE REVIEW

File: src/Util.ts [hunk 1]
```diff
@@ -420,6 +420,25 @@ class Util {
...
+        try {
+            const jsonData = html.split('{"horizontalCardListRenderer":')[1].split(',{"reelShelfRenderer"')[0]
+
+            musicInfo = JSON.parse('{"horizontalCardListRenderer":' + jsonData).horizontalCardListRenderer.cards.map((val: any) => {
+                return {
+                    title: val.videoAttributeViewModel.title,
+                    cover: val.videoAttributeViewModel.image.sources[0].url,
+                    artist: val.videoAttributeViewModel.subtitle,
+                    album: val.videoAttributeViewModel.secondarySubtitle.content,
+                    raw: val.videoAttributeViewModel
```
Comment (twlite) [inline src/Util.ts:435]: I don't think we need to expose raw data here

**Review by twlite — CHANGES_REQUESTED** [review]

File: src/Util.ts [hunk 2]
```diff
@@ -450,7 +469,8 @@ class Util {
                 dislikes: 0
             },
             videos: Util.getNext(nextData ?? {}) || [],
-            streamingData: info.info.streamingData || null
+            streamingData: info.info.streamingData || null,
+            music: musicInfo
```
Comment (twlite) [inline src/Util.ts:473]: do we call it `metadata`?

File: src/Util.ts (hunk 1 above)
Comment (retrouser955) [reply src/Util.ts:435]: Ok, I'll change it!

File: src/Util.ts (hunk 2 above)
Comment (retrouser955) [reply src/Util.ts:473]: metadata seems too vague I feel like.

Comment (twlite) [reply src/Util.ts:473]: ok

File: src/Structures/Video.ts [hunk 3]
```diff
@@ -83,6 +90,7 @@ export class Video {
     shorts = false;
     unlisted = false;
     streamingData?: VideoStreamingData | null;
+    music: MusicInfo[] | void
```
Comment (twlite) [inline src/Structures/Video.ts:93]:
```suggestion
    music: MusicInfo[] = [];
```

File: src/Util.ts [hunk 4]
```diff
@@ -420,6 +420,24 @@ class Util {
             info
         };
 
+        // Get music info if there are any
+        let musicInfo: MusicInfo[] | void;
```
Comment (twlite) [inline src/Util.ts:424]:
```suggestion
        let musicInfo: MusicInfo[] = [];
```

File: src/Util.ts [hunk 5]
```diff
@@ -420,6 +420,24 @@ class Util {
...
+            musicInfo = JSON.parse('{"horizontalCardListRenderer":' + jsonData).horizontalCardListRenderer.cards.map((val: any) => {
+                return {
+                    title: val.videoAttributeViewModel.title,
+                    cover: val.videoAttributeViewModel.image.sources[0].url,
+                    artist: val.videoAttributeViewModel.subtitle,
+                    album: val.videoAttributeViewModel.secondarySubtitle.content
+                }
+            })
+        } catch {
+            musicInfo = undefined
```
Comment (twlite) [inline src/Util.ts:438]:
```suggestion
```

**Review by twlite — CHANGES_REQUESTED** [review]

**Review by twlite — APPROVED** [review]

**Pull request merged**